Core/Interfaces.md

```diff
@@ -1,6 +1,6 @@
 Anything that acts as a common interface between sets of disparate entities.
 
-The sets of disparate entities may be composed of competitive alternatives and/or cooperative elements.
+The sets of disparate entities may be factorized of competitive alternatives (sum types? https://en.wikipedia.org/wiki/Tagged_union) and/or cooperative elements (product types?).
```

Comment:
I prefer "composed" here, as "factorized" is more related to the opposite direction: breaking a whole into parts.

I'm not sure sum/product types are all that useful an analogy here.